README.mkd

```diff
@@ -207,6 +207,27 @@ RV32E. There is not currently a way to specify an alternative stack alignment,
 but the `-mpreferred-stack-boundary` and `-mincoming-stack-boundary` flags 
 supported by GCC on X86 could be adopted.
 
+## Save restore support
+
+The save restore optimization is enabled through the option `-msave-restore`
+and reduces the amount of code in the prologue and epilogue by using
+library functions instead of inline code to save and restore callee saved
+registers. The library functions are provided in the emulation library and
+have the following signatures:
+
+* `void __riscv_save_<N>(void)`
+* `void __riscv_restore_<N>(void)`
+
+`<N>` is a value between 0 and 12 and corresponds to the number of
+registers between `s0` and `s11` that are saved/restored. The return
+address register `ra` is always included in the registers saved and restored.
+
+The `__riscv_save_<N>` functions are called from the prologue. They allocate
```

Comment:
The save function call must use the alternate link register t0 instead of the regular ra, to avoid clobbering ra.  The save function then returns to the address in t0.